### PR TITLE
VPC flags and security group blocs using VPC cidrs to allow the test controller to use VPC resources provisioned outside the module

### PR DESCRIPTION
**Pull request by bvkin**
Signed-off-by: Benjamin Kincaid <ben.kincaid@rocketinsights.com>

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -12,20 +17,112 @@ variable "ec2_public_key" {
 }
 
 # VPC Variables
+variable "create_vpcs" {
+  type = string
+  description = "Flag to VPCs and related resources"
```
Comment (kylecrawshaw) [inline variables.tf]: Missing `create`

File: variables.tf [hunk 2]
```diff
@@ -12,20 +17,112 @@ variable "ec2_public_key" {
 }
 
 # VPC Variables
+variable "create_vpcs" {
```
Comment (kevinkarwaski) [inline variables.tf]: I think `create_networking` maybe works better?

File: variables.tf [hunk 3]
```diff
@@ -12,20 +17,112 @@ variable "ec2_public_key" {
...
+  default = null
+}
+variable "vpc_azs_usw2" {
+  type = list(string)
+  description = "AZs of VPC in us-east-2"
+  default = null
+}
+variable "s3_interface_endpoint_use1" {
+  type = string
+  description = "S3 endpoint for VPC in us-east-1"
```
Comment (kylecrawshaw) [inline variables.tf]: I think we need a default here?

File: variables.tf (hunk 1 above)
Comment (kevinkarwaski) [inline variables.tf]: `Flag to create VPCs and related resources`

**Review by kevinkarwaski — CHANGES_REQUESTED** [review]

File: modules/test-controller/variables.tf [hunk 4]
```diff
@@ -4,6 +4,12 @@ variable "vpc_id" {
   default     = ""
 }
 
+variable "vpc_cidr_blocks" {
+  description = "A list of VPC cidr blocks to add to the interface enpoint security group"
+  type = list(string)
+  default = []
```
Comment (kevinkarwaski) [inline modules/test-controller/variables.tf]: Set the default to [`10.0.0.0/8`]

File: variables.tf [hunk 5]
```diff
@@ -12,20 +17,115 @@ variable "ec2_public_key" {
...
+variable "s3_interface_endpoint_usw2" {
+  type = string
+  description = "S3 endpoint for VPC in us-west-2"
+  default = null
+}
 variable "use1_main_network_block" {
   type        = string
   description = "Base CIDR block to be used in us-east-1."
-  default     = "10.0.0.0/16"
+  default     = null
```
Comment (kevinkarwaski) [inline variables.tf]: Re-add the default blocks here so default behavior will work without setting any inputs

**Review by kevinkarwaski — CHANGES_REQUESTED** [review]

**Review by kevinkarwaski — APPROVED** [review]

**Pull request merged**